trade-finance-manager-ui/server/controllers/case/cancellation/validation/validate-bank-request-date.test.ts

```diff
@@ -0,0 +1,100 @@
+import { validateBankRequestDate } from './validate-bank-request-date';
+import { BankRequestDateValidationViewModel } from '../../../../types/view-models';
+
+describe('validateBankRequestDate', () => {
+  const today = startOfDay(new Date());
+
+  const validDates = [
+    {
+      description: 'today',
+      date: today,
+    },
+    {
+      description: `11 months in the future`,
+      date: add(today, { months: 11 }),
+    },
+    {
+      description: `11 months in the past`,
+      date: add(today, { months: -11 }),
+    },
+  ];
+
+  it.each(validDates)('returns no errors when the date is $description', ({ date }) => {
+    // Act
+    const day = date.getDate().toString();
+    const month = (date.getMonth() + 1).toString();
+    const year = date.getUTCFullYear().toString();
+
+    const result = validateBankRequestDate({ day, month, year });
+
+    // Assert
+    const expected: BankRequestDateValidationViewModel = {
+      errors: null,
+      bankRequestDate: date,
+    };
+    expect(result).toEqual(expected);
+  });
+
+  it('returns an error if the bank request date is greater than 12 months in the future', () => {
+    const invalidBankRequestDate = add(new Date(), { months: 12, days: 1 });
+    const day = invalidBankRequestDate.getDate().toString();
+    const month = (invalidBankRequestDate.getMonth() + 1).toString();
+    const year = invalidBankRequestDate.getFullYear().toString();
+
+    const result = validateBankRequestDate({ day, month, year });
+
+    const expectedMessage = 'The bank request date cannot exceed 12 months in the future from the submission date';
+
+    const expectedError = {
+      summary: [{ text: expectedMessage, href: '#bank-request-date-day' }],
+      bankRequestDateError: { message: expectedMessage, fields: ['bank-request-date-day', 'bank-request-date-month', 'bank-request-date-year'] },
+    };
+
+    expect(result.errors).toEqual(expectedError);
+  });
+
+  it('returns an error if the bank request date is greater than 12 months in the past', () => {
+    const invalidBankRequestDate = add(new Date(), { months: -12, days: -1 });
+    const day = invalidBankRequestDate.getDate().toString();
+    const month = (invalidBankRequestDate.getMonth() + 1).toString();
+    const year = invalidBankRequestDate.getFullYear().toString();
+
+    const result = validateBankRequestDate({ day, month, year });
+
+    const expectedMessage = 'The bank request date cannot exceed 12 months in the past from the submission date';
+
+    const expectedError = {
+      summary: [{ text: expectedMessage, href: '#bank-request-date-day' }],
+      bankRequestDateError: { message: expectedMessage, fields: ['bank-request-date-day', 'bank-request-date-month', 'bank-request-date-year'] },
+    };
+
+    expect(result.errors).toEqual(expectedError);
+  });
+
+  it('returns an error if a field is left blank', () => {
+    const result = validateBankRequestDate({ day: '', month: '1', year: '2024' });
```

Comment:
Can we add some tests for if `month` or `year` are left empty?

Reply:
Thanks @BethThomas141 makes sense, I don't think we should have duplicate tests like you say.

I do think we need to ensure that in here, or any other function that calls/consumes `applyStandardValidationAndParseDateInput`, need to make sure that it is called - not sure what's in place already but maybe something like this would suffice?

```ts
describe('X field validation', () => {
  describe('custom date rules', () => {
    it('should ...'
    it('should ...'
  });

  describe('when all custom date rules are valid', () => {
    it('should return the result of applyStandardValidationAndParseDateInput', () => {
      const expected = applyStandardValidationAndParseDateInput();

      expect(result).toEqual(expected);
    });
  });
});
```

Reply:
Good idea, thanks! I've now done something to this effect - have a look and let me know what you think.

Reply:
Thanks @BethThomas141 looks good to me 

Appreciate this isn't part of DTFS2-7296 - are we able to check and apply the same treatment to other date unit tests that use `applyStandardValidationAndParseDateInput`, so they're all aligned? Not sure if this would be a quick thing to do

Reply:
I think we leave it for now as it's out of scope of this bit of work, but certainly going forwards we can

Reply:
Can we create a ticket for this please? So we don't forget to make it aligned with the nice new approach we have here 😊

Reply:
It looks like we have tests that would be impacted in only 2 files - so once we've nailed down the approach in this PR, I'll do a PR straight away afterwards this afternoon to amend those other two (but will keep it out of this PR).